cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesOnDataSource.java

```diff
@@ -37,8 +37,11 @@ public ReadOnlyDataSource dataSource() {
     private boolean checkIfMainFileNotWithCgmesData(boolean isCim14) {
         if (dataSource.getDataExtension() == null || dataSource.getDataExtension().isEmpty()) {
             return false;
-        } else if (EXTENSIONS.equals(dataSource.getDataExtension())) {
-            try (InputStream is = dataSource.newInputStream(null, EXTENSIONS)) {
+        } else if (EXTENSION.equals(dataSource.getDataExtension())) {
+            try (InputStream is = dataSource.exists(null, EXTENSION) ? dataSource.newInputStream(null, EXTENSION) : null) {
+                if (is == null) {
+                    return true;
+                }
```

Comment:
why not checking beforehand?
```suggestion
            if (dataSource.exists(null, EXTENSION) == null) {
                return true;
            }
            try (InputStream is = dataSource.newInputStream(null, EXTENSION)) {
```

Reply:
because datasource.exists is declared to throw IOexception, so we need to rewrap with an extra try catch

Reply:
Well I'd rather have 2 nested try than current code. Or we could catch the `IOException` only in `CgmesImport::exists`
```suggestion
            try {
                if (!dataSource.exists(null, EXTENSION)) {
                    return true;
                }
                try (InputStream is = dataSource.newInputStream(null, EXTENSION)) {
```